packages/sanity/src/core/form/components/formField/FormFieldHeader.tsx

```diff
@@ -1,75 +1,69 @@
 /* eslint-disable camelcase */
-
 import React, {memo} from 'react'
-import {Box, Flex} from '@sanity/ui'
 import {FormNodeValidation} from '@sanity/types'
-import styled from 'styled-components'
-import {FieldPresence, FormNodePresence} from '../../../presence'
+import {FormNodePresence} from '../../../presence'
+import {DocumentFieldActionNode} from '../../../config'
 import {FormFieldHeaderText} from './FormFieldHeaderText'
+import {FormFieldBaseHeader} from './FormFieldBaseHeader'
+
+const EMPTY_ARRAY: never[] = []
 
 export interface FormFieldHeaderProps {
   /**
    *
    * @hidden
    * @beta
    */
-  __unstable_actions?: React.ReactNode
-  /**
-   * @beta
-   */
-  validation?: FormNodeValidation[]
+  __unstable_actions?: DocumentFieldActionNode[]
   /**
    *
    * @hidden
    * @beta
    */
   __unstable_presence?: FormNodePresence[]
+  /**
+   * @beta
+   */
+  validation?: FormNodeValidation[]
   description?: React.ReactNode
   /**
    * The unique ID used to target the actual input element
    */
   inputId?: string
   title?: React.ReactNode
-}
 
-const Root = styled(Flex)({
-  // This prevents the buttons from taking up extra vertical space in the flex layout,
-  // due to their default vertical alignment being baseline.
-  lineHeight: 1,
-})
+  /** @hidden */
+  fieldHovered: boolean
+  /** @hidden */
+  fieldFocused: boolean
+}
 
 export const FormFieldHeader = memo(function FormFieldHeader(props: FormFieldHeaderProps) {
```

Comment:
**question**: is `FormFieldHeader` being imported anywhere? From what I can gather it isn't – if so, we should consider removing it

Reply:
I'll remove it 👍